### PR TITLE
Add an example to show how to insert item to a sorted vec

### LINKED ISSUE #61684: `binary_search` docs should show how to insert something into an already-sorted list, while maintaining sort order
This isn't exactly hard to do, but it's completely possible to have an off-by-one error because of overthinking it, and it's a common enough use case that it seems worth showing off in a doctest.

### PR DESCRIPTION
**Pull request by tesuji**
Closes #61684
cc #61742
r? @Mark-Simulacrum, @jonas-schievink

### CODE REVIEW

File: src/libcore/slice/mod.rs [hunk 1]
```diff
@@ -1364,6 +1364,20 @@ impl<T> [T] {
...
+    ///
+    /// If you want to insert an item to a sorted vector, while maintaining
+    /// sort order:
+    ///
+    /// ```
+    /// let mut s = vec![0, 1, 1, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55];
+    /// let num = 42;
+    /// let idx = match s.binary_search(&num) {
+    ///     Ok(idx) => idx,
+    ///     Err(idx) => idx,
```
Comment (jonas-schievink) [inline src/libcore/slice/mod.rs]: The arms could be simplified by using `Ok(idx) | Err(idx) => idx`

Comment (Mark-Simulacrum) [reply src/libcore/slice/mod.rs]: I personally find this clearer than that but r=me with this changed as well.

Comment (Centril) [reply src/libcore/slice/mod.rs]: This should be further simplified to `let Ok(idx) | Err(idx) = s.binary_search(&num);` once we have or-patterns, https://github.com/rust-lang/rust/issues/54883.

Comment (lzutao) [reply src/libcore/slice/mod.rs]: I use `unwrap_or_else`. Do you prefer it instead?

**Comment by Mark-Simulacrum** [comment]
The identity closure always irks me but this works too. @bors r+

**Pull request merged**